draft-ietf-tls-esni.md

```diff
@@ -885,10 +885,13 @@ version was negotiated, the client can regard ECH as securely disabled
 by the server, and it SHOULD retry the handshake with a new transport
 connection and ECH disabled.
 
-Clients SHOULD implement a limit on retries caused by receipt of "retry_configs"
-or servers which do not acknowledge the "encrypted_client_hello" extension. If
-the client does not retry in either scenario, it MUST report an error to the
-calling application.
+Clients SHOULD NOT accept "retry_config" in response to a connection
+initiated in response to a "retry_config".  Sending a "retry_config"
+in this situation is a signal that the server is misconfigured, e.g.,
+the server might have multiple inconsistent configurations so that the
+client reached a node with configuration A in the first connection and
+a node with configuration B in the second. If a client does not retry,
```

Comment:
The previous paragraph also talks about a "retry". But that is an ECH *disabled* retry.
In this paragraph, based on the context, it is referring to an ECH *enabled* retry.
Should the client be allowed to perform an ECH disabled retry here? And should anything about that, allowed or not allowed, be noted in this paragraph also?

Reply:
I think this is a reasonable question. Updated the text.

Note that I also broke out the requirement to report the error separately as I think it applies to both cases.